.github/workflows/compile.yml

```diff
@@ -0,0 +1,22 @@
+name: compile test
+
+on: [push, pull_request]
```

Comment:
I don't think we want this on pushes to dev branches as they are often broken. I normally do this on PR against master and then also make it part of the master branch rules.